src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/BlackDuckBomEditCollector.java

```diff
@@ -0,0 +1,340 @@
+import com.synopsys.integration.blackduck.api.generated.view.RiskProfileView;
+import com.synopsys.integration.blackduck.api.generated.view.VersionBomComponentView;
+import com.synopsys.integration.blackduck.api.generated.view.VulnerabilityWithRemediationView;
+import com.synopsys.integration.blackduck.api.generated.view.VulnerableComponentView;
+import com.synopsys.integration.blackduck.service.ComponentService;
+import com.synopsys.integration.blackduck.service.model.ProjectVersionWrapper;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.log.Slf4jIntLogger;
+
+@Component
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class BlackDuckBomEditCollector extends BlackDuckCollector {
+    private static final String CATEGORY_TYPE_REMEDIATION = "Vulnerability Remediation";
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    public BlackDuckBomEditCollector(final JsonExtractor jsonExtractor, final List<MessageContentProcessor> messageContentProcessorList, final BlackDuckProperties blackDuckProperties) {
+        super(jsonExtractor, messageContentProcessorList, List.of(BlackDuckContent.BOM_EDIT), blackDuckProperties);
+    }
+
+    @Override
+    protected List<LinkableItem> getTopicItems(final JsonFieldAccessor accessor, final List<JsonField<?>> fields) {
+        final JsonField<String> topicField = getDataField(fields, FieldContentIdentifier.TOPIC);
+        final Optional<String> bomComponentUrl = getBomComponentUrl(accessor, topicField);
+        final Optional<ProjectVersionWrapper> projectVersionWrapper = bomComponentUrl
+                                                                          .flatMap(this::getBomComponentView)
+                                                                          .flatMap(this::getProjectVersionWrapper);
+        if (projectVersionWrapper.isPresent()) {
+            ProjectView projectView = projectVersionWrapper.get().getProjectView();
+            return List.of(new LinkableItem(topicField.getLabel(), projectView.getName(), projectView.getHref().orElse(null)));
+        }
+
+        return List.of();
+    }
+
+    @Override
+    protected List<LinkableItem> getSubTopicItems(final JsonFieldAccessor accessor, final List<JsonField<?>> fields) {
+        final JsonField<String> topicField = getDataField(fields, FieldContentIdentifier.SUB_TOPIC);
+        final Optional<String> bomComponentUrl = getBomComponentUrl(accessor, topicField);
+        final Optional<ProjectVersionWrapper> projectVersionWrapper = bomComponentUrl
+                                                                          .flatMap(this::getBomComponentView)
+                                                                          .flatMap(this::getProjectVersionWrapper);
+        if (projectVersionWrapper.isPresent()) {
+            ProjectVersionView projectVersionView = projectVersionWrapper.get().getProjectVersionView();
```

Comment:
This could be taking a step further in your mapping.
`.flatMap(ProjectVersionWrapper::getProjectVersionView)`

Reply:
The project version view isn't optional.